frontend/src/pages/RunDetails.tsx

```diff
@@ -877,30 +905,52 @@ class RunDetails extends Page<RunDetailsInternalProps, RunDetailsState> {
 
   private async _loadSelectedNodeLogs(): Promise<void> {
     const selectedNodeDetails = this.state.selectedNodeDetails;
-    if (!selectedNodeDetails) {
+    const namespace = this.state.workflow?.metadata?.namespace;
+    if (!selectedNodeDetails || !namespace) {
       return;
     }
     this.setStateSafe({ sidepanelBusy: true });
+
+    let logsBannerMessage = '';
+    let logsBannerAdditionalInfo = '';
+    let logsBannerMode = '' as Mode;
+
     try {
-      const logs = await Apis.getPodLogs(
-        selectedNodeDetails.id,
-        this.state.workflow?.metadata?.namespace,
-      );
-      selectedNodeDetails.logs = logs;
+      selectedNodeDetails.logs = await Apis.getPodLogs(selectedNodeDetails.id, namespace);
+    } catch (err) {
+      let errMsg = await errorToMessage(err);
+      logsBannerMessage = 'Failed to retrieve pod logs.';
+
+      // if pod can't be found, then assume it has gone from the server
+      if (await this.podNotFound(selectedNodeDetails.id, namespace)) {
+        logsBannerMessage += this.props.gkeMetadata.projectId
+          ? ' Use Stackdriver Kubernetes Monitoring to view them.'
+          : '';
+        logsBannerMode = 'info';
+        logsBannerAdditionalInfo =
+          'Possible reasons include cluster autoscaling, pod garbage collection and pod preemption. ';
+      } else {
+        logsBannerMode = 'error';
+      }
+
+      logsBannerAdditionalInfo += 'Error response: ' + errMsg;
+    } finally {
```

Comment:
nit: maybe drop the finally? It seems unnecessary, because a normal code block after the try catch would suit our case too.